wallets/endpoint.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (C) 2022 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package wallets
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"storj.io/storjscan/tokens"
+)
+
+// ErrEndpoint is the wallets endpoint error class.
+var ErrEndpoint = errs.Class("Wallets Endpoint")
+
+// Endpoint for interacting with the Wallets service.
+//
+// architecture: Endpoint
+type Endpoint struct {
+	log     *zap.Logger
+	service Wallets
+}
+
+// NewEndpoint creates new wallets endpoint instance.
+func NewEndpoint(log *zap.Logger, wallets Wallets) *Endpoint {
+	return &Endpoint{
+		log:     log,
+		service: wallets,
+	}
+}
+
+// Register registers endpoint methods on API server subroute.
+func (endpoint *Endpoint) Register(router *mux.Router) {
+	router.HandleFunc("/wallets", endpoint.GetNewDepositAddress).Methods(http.MethodGet)
```

Comment:
As this method modifies the database, I would consider to use `http.MethodPut` or `http.MethodPost` to avoid attack like linking this endpoint as an image which can exhaust the pool.

Reply:
good thinking!